resolver/addressable_resolver.go

```diff
@@ -38,16 +38,21 @@ import (
 	"knative.dev/pkg/tracker"
 )
 
+type URIResolverFunc func(ctx context.Context, ref *corev1.ObjectReference) (*apis.URL, error)
+
 // URIResolver resolves Destinations and ObjectReferences into a URI.
 type URIResolver struct {
 	tracker       tracker.Interface
 	listerFactory func(schema.GroupVersionResource) (cache.GenericLister, error)
+	resolvers     []URIResolverFunc
 }
 
 // NewURIResolver constructs a new URIResolver with context and a callback
 // for a given listableType (Listable) passed to the URIResolver's tracker.
```

Comment:
Please add documentation for the new argument `resolvers`.

Reply:
done!